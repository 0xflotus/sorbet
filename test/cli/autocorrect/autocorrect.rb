# typed: strict

foo = T.let(nil, T.nilable(String))
foo[0]

"hi" + foo

sig(a: String).void
def int(a:); end
int(a: foo)

foo.bar ||= "a"
foo.bar &&= "a"
