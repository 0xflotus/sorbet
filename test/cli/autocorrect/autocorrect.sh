tmp=$(mktemp)
file=test/cli/autocorrect/autocorrect.rb
cp "$file" "$tmp"
main/sorbet -a "$tmp"
diff "$file" "$tmp"
rm "$tmp"
