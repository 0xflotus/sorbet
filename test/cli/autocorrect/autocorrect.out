4c4
< foo[0]
---
> T.must(foo)[0]
6c6
< "hi" + foo
---
> "hi" + T.must(foo)
12,15c12,15
< foo.bar ||= "a"
< foo.bar &&= "a"
< foo.bar |= "a"
< foo.bar &= "a"
---
> T.must(foo.bar) ||= "a"
> T.must(foo.bar) &&= "a"
> T.must(foo.bar) |= "a"
> T.must(foo.bar) &= "a"
